lldb/tools/lldb-dap/tool/lldb-dap.cpp

```diff
@@ -341,26 +332,11 @@ serveConnection(const Socket::SocketProtocol &protocol, const std::string &name,
       log,
       "lldb-dap server shutdown requested, disconnecting remaining clients...");
 
-  bool client_failed = false;
-  {
-    std::scoped_lock<std::mutex> lock(dap_sessions_mutex);
-    for (auto [sock, dap] : dap_sessions) {
-      if (llvm::Error error = dap->Disconnect()) {
-        client_failed = true;
-        llvm::errs() << "DAP client " << dap->transport.GetClientName()
-                     << " disconnected failed: "
-                     << llvm::toString(std::move(error)) << "\n";
-      }
-    }
-  }
```

Comment:
I know you are doing a lot of refactoring to existing code into DAPSessionManager class. I would suggest you create a separate PR into upstream main to upstream the change. That would avoid potential merge conflicts in future considering several folks are actively refactoring/changing this in upstream.